lib/elixir/src/elixir_erl.erl

```diff
@@ -329,8 +329,8 @@ macros_info(Defmacro) ->
   {clause, 0, [{atom, 0, macros}], [], [elixir_erl:elixir_to_erl(lists:sort(Defmacro))]}.
 
 get_module_info(Module, Key) ->
-  Call = remote(0, erlang, get_module_info, [{atom, 0, Module}, {atom, 0, Key}]),
-  {clause, 0, [{atom, 0, Key}], [], [Call]}.
+  Call = remote(0, erlang, get_module_info, [{atom, 0, Module}, {var, 0, 'Key'}]),
```

Comment:
I think it might be a good idea to include some comments like the ones you gave on https://github.com/elixir-lang/elixir/pull/7753#issuecomment-397270877 :)

I think this change might not be so obvious for future Elixir contributors.